sklearn/linear_model/tests/test_passive_aggressive.py

```diff
@@ -125,6 +126,77 @@ def test_classifier_undefined_methods():
         assert_raises(AttributeError, lambda x: getattr(clf, x), meth)
 
 
+def test_class_weights():
+    # Test class weights.
+    X2 = np.array([[-1.0, -1.0], [-1.0, 0], [-.8, -1.0],
+                   [1.0, 1.0], [1.0, 0.0]])
+    y2 = [1, 1, 1, -1, -1]
+
+    clf = PassiveAggressiveClassifier(C=0.1, n_iter=100, class_weight=None,
+                                      random_state=100)
+    clf.fit(X2, y2)
+    assert_array_equal(clf.predict([[0.2, -1.0]]), np.array([1]))
+
+    # we give a small weights to class 1
+    clf = PassiveAggressiveClassifier(C=0.1, n_iter=100,
```

Comment:
I'm not entirely convinced it's better, but I can see some reasons for just doing `clf.set_params(class_weight={1: 0.001})` here. It makes it explicit that the rest of the parameters shouldn't be changed, in case somebody modifies the test in the future.